backend/app/processing/kb.py

```
@@ -0,0 +1,296 @@
+    # Collect all supported files under both directories
+    base_dir = os.path.dirname(os.path.abspath(__file__))
+    kb_dir = os.path.join(base_dir, "../knowledge_bases", chatbot_id)
+    scrapped_dir = os.path.join(base_dir, "../scrapped_files", chatbot_id)
+    file_paths = []
+    
+    for root_dir in (kb_dir, scrapped_dir):
+        if not os.path.exists(root_dir):
+            continue
+        for root, _, files in os.walk(root_dir):
+            for fname in files:
+                if fname.startswith('.') or fname.startswith('~$'):
```

Ruff (0.8.2)

Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)